sklearn/pipeline.py

```diff
@@ -530,6 +546,66 @@ def _pairwise(self):
         # check if first estimator expects pairwise input
         return getattr(self.steps[0][1], '_pairwise', False)
 
+    def set_feature_names(self, input_features):
+        """Set the input feature names for all steps.
+
+        Sets the input_features_ attribute on the pipeline and
+        on all pipeline steps using the provided input feature names
+        as input for the first step.
+
+        Some estimators like `ColumnTransformer` and `CountVectorizer`
+        might ignore the provided input feature names.
+
+        Parameters
+        ----------
+        input_features : array-like of string or None
+            Feature names to use as input feature names for the first step
+            of the pipeline.
+
+        """
+        self.input_features_ = input_features
+        feature_names = input_features
+        for _, name, transform in self._iter(with_final=False):
+            transform.input_features_ = feature_names
+            if not hasattr(transform, "get_feature_names"):
+                raise TypeError("Transformer {} does provide"
+                                " get_feature_names".format(name))
+            try:
+                feature_names = transform.get_feature_names(
+                    input_features=feature_names)
+            except TypeError:
+                feature_names = transform.get_feature_names()
+        self._final_estimator.input_features_ = feature_names
+
+    def get_feature_names(self, input_features=None):
+        """Get feature names for transformation.
+
+        Transform input features using the pipeline.
+        If the last step is a transformer, it's included
+        in the transformation, otherwise it's not.
+
+        Parameters
+        ----------
+        input_features : array-like of string
+            Input feature names.
+
+        Returns
+        -------
+        feature_names : array-like of string
+            Transformed feature names
+        """
+        feature_names = input_features
```

Comment:
if input_features is None, take the one that might already be stored on `self` ?

Reply:
done.